### ISSUE TITLE
Error : Doesn't work anymore

### ISSUE THREAD

**Issue by Danavans** [RileyXX/TMDB-Trakt-Syncer#33]
### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Bug Description

Hi,
It was working perfectly fine for a long time, but since few months I have this error every time I start a sync, and can't understand why. I tried to reinstall the script, but nothing change, still this error. Thanks for your help.

   - Request failed with exception: ('Connection aborted.', ConnectionResetError(10054, 'Une connexion existante a dû être fermée par l’hôte distant', None, 10054, None))

--- ERROR ---
An error occurred while running the script.
Please submit the error to GitHub with the following information:
--------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\xxxx\AppData\Local\Programs\Python\Python311\Lib\site-packages\TMDBTraktSyncer\TMDBTraktSyncer.py", line 22, in main
    tmdb_watchlist, tmdb_ratings = tmdbData.getTMDBRatings()
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\xxxx\AppData\Local\Programs\Python\Python311\Lib\site-packages\TMDBTraktSyncer\tmdbData.py", line 24, in getTMDBRatings
    json_data = json.loads(response.text)
                           ^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'text'

### Environment

OS Windows 11
Install method using pip

**Comment by RileyXX** [comment]
Hi thanks for reporting this. This could potentially be an issue where credentials from an older version were not updated properly.

First make sure you're on the latest version by using `python -m pip install TMDBTraktSyncer --upgrade`. Then to reset your credentials delete the `credentials.txt` file in your scripts folder which would be located at `C:\Users\xxxx\AppData\Local\Programs\Python\Python311\Lib\site-packages\TMDBTraktSyncer\`. Then run the script again and it should prompt you to re-enter your credentials. See if that resolves this issue. If not let me know and I'll take a deeper look.

**Comment by Danavans** [comment]
Hi,
Thanks for your reply, and sorry for the late mine.
I just did what you asked me to do, but nothing change, I still have the same error : 


Processing Trakt Data
Processing Trakt Data Complete
Processing TMDB Data
   - Request failed with exception: ('Connection aborted.', ConnectionResetError(10054, 'Une connexion existante a dû être fermée par l’hôte distant', None, 10054, None))

--- ERROR ---
An error occurred while running the script.
Please submit the error to GitHub with the following information:
--------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\xxxxx\AppData\Local\Programs\Python\Python311\Lib\site-packages\TMDBTraktSyncer\TMDBTraktSyncer.py", line 22, in main
    tmdb_watchlist, tmdb_ratings = tmdbData.getTMDBRatings()
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\xxxxx\AppData\Local\Programs\Python\Python311\Lib\site-packages\TMDBTraktSyncer\tmdbData.py", line 24, in getTMDBRatings
    json_data = json.loads(response.text)
                           ^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'text'

--------------------------------------------------
Submit the error here: https://github.com/RileyXX/TMDB-Trakt-Syncer/issues/new?template=bug_report.yml
--------------------------------------------------

I hope you can find the source of the problem.
Thanks for your help.

**Comment by sreyaz01** [comment]
````````````````````````````````````````````````````````````````````````````````````````````````````
2024-05-08 14:59:13,070 - ERROR - Request failed with status code 401: This resource is private
````````````````````````````````````````````````````````````````````````````````````````````````````

````````````````````````````````````````````````````````````````````````````````````````````````````
2024-05-08 14:59:13,086 - ERROR - An error occurred while running the script.
Traceback (most recent call last):
  File "D:\Downloads\TMDBTraktSyncer\TMDBTraktSyncer\TMDBTraktSyncer.py", line 22, in main
    tmdb_watchlist, tmdb_ratings = tmdbData.getTMDBRatings()
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Downloads\TMDBTraktSyncer\TMDBTraktSyncer\tmdbData.py", line 24, in getTMDBRatings
    json_data = json.loads(response.text)
                           ^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'text'
````````````````````````````````````````````````````````````````````````````````````````````````````

i think problem is in this block of code. 

![image](https://github.com/RileyXX/TMDB-Trakt-Syncer/assets/13959463/e0d9cf12-15b9-48e2-8837-d503e3da19e0)

**Comment by kvl19** [comment]
> ```
> 2024-05-08 14:59:13,070 - ERROR - Request failed with status code 401: This resource is private
> ```
> 
> ```
> 2024-05-08 14:59:13,086 - ERROR - An error occurred while running the script.
> Traceback (most recent call last):
>   File "D:\Downloads\TMDBTraktSyncer\TMDBTraktSyncer\TMDBTraktSyncer.py", line 22, in main
>     tmdb_watchlist, tmdb_ratings = tmdbData.getTMDBRatings()
>                                    ^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "D:\Downloads\TMDBTraktSyncer\TMDBTraktSyncer\tmdbData.py", line 24, in getTMDBRatings
>     json_data = json.loads(response.text)
>                            ^^^^^^^^^^^^^
> AttributeError: 'NoneType' object has no attribute 'text'
> ```
> 
> i think problem is in this block of code.
> 
> ![image](https://private-user-images.githubusercontent.com/13959463/328842631-e0d9cf12-15b9-48e2-8837-d503e3da19e0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwNjA1NzAsIm5iZiI6MTcyNTA2MDI3MCwicGF0aCI6Ii8xMzk1OTQ2My8zMjg4NDI2MzEtZTBkOWNmMTItMTViOS00OGUyLTg4MzctZDUwM2UzZGExOWUwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMwVDIzMjQzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEzMzA2Yjg2ZWY5ZTk1NjU0MjhkYmQzNzdkZTJkNWI2MDNjYWE0NTlkYmRjNjZmODdiZjg0M2FiOWUyNWY2MWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1T2GUwwnpUexHSqszF0AAkj_kRne95lYmYG-DmaXWZ8)

I got the same error and I think the issue is because my tmdb_access_token expired or changed.
So I went to https://www.themoviedb.org/settings/api/ and regenerate the key again and update that API Read Access Token in credentials.txt, so you can just delete credentials.txt to set everything up again

**Comment by RileyXX** [comment]
Hi sorry for the late reply. This issue should be fixed in the latest version v1.7.3 or later. A problem was identified in another project where the script was improperly handling network errors such as error 54 connection reset by server (rate limiting). If you are still having problems with this let me know and I can reopen this issue. Thanks
